rotkehlchen/db/updates.py

```diff
@@ -0,0 +1,113 @@
+        self.user_db = user_db
+        self.branch = 'main'
+        if not getattr(sys, 'frozen', False):
+            # not packaged -- must be in develop mode
+            self.branch = 'develop'
+
+    def _get_remote_info_json(self) -> dict[str, Any]:
+        """Retrieve remote file with information for different updates"""
+        url = f'https://raw.githubusercontent.com/rotki/data/{self.branch}/updates/info.json'
+        try:
+            response = requests.get(url=url, timeout=DEFAULT_TIMEOUT_TUPLE)
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Failed to query {url} during assets update: {str(e)}') from e
+
+        try:
+            json_data = response.json()
+        except json.decoder.JSONDecodeError as e:
+            raise RemoteError(
+                f'Could not parse update info from {url} as json: {response.text}',
+            ) from e
+
+        return json_data
+
+    def update_spam_assets(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        spam assets for those versions. Assets are also added to the globaldb if they don't exist
+        locally
+        """
+        for version in range(local_version + 1, to_version + 1):
+            file_url = SPAM_ASSETS_URL.format(branch=self.branch, version=version)
+            try:
+                updates = query_file(file_url, is_json=True)
+            except RemoteError as e:
+                log.warning(f'Failed to update spam assets due to {str(e)}')
+                break
+            try:
+                if len(updates['assets']) != 0:
+                    log.info(f'Applying update for spam assets from {version}. {len(updates)} tokens to add')  # noqa: E501
+                    with GlobalDBHandler().conn.critical_section():
+                        # Use a critical section to avoid another greenlet adding spam assets at
+                        # the same time
+                        update_spam_assets(db=self.user_db, assets_info=updates['assets'])
+            except KeyError as e:
+                log.error(f'Remote update {file_url} does not contain the expected key {str(e)}')
+                continue
+
+            with self.user_db.conn.write_ctx() as write_cursor:
+                write_cursor.execute(
+                    'INSERT OR REPLACE INTO settings(name, value) VALUES (?, ?)',
+                    (UpdateType.SPAM_ASSETS.serialize(), version),
+                )
+
+    def check_for_updates(self) -> None:
+        """Retrieve the information about the latest available update"""
+        remote_information = self._get_remote_info_json()
+
+        # Get latest applied versions
+        with self.user_db.conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT value FROM settings WHERE name=?',
+                (UpdateType.SPAM_ASSETS.serialize(),),
```

Comment:
nope. Man ... think a bit more. Don't you see what this will do? This will add a setting with name `spam assets` in the DB? Really? You think this is a good idea?

Just override serialize and don't diretly use `SerializableEnumValueMixin`. I suggested it only as a way to see how an enum can be serialized. 

I will leave a comment in the enum itself to show you what you should do.